### DISCUSSION TITLE
Feature branches

### DISCUSSION THREAD

**Discussion by JZWong** [ProjectBorealis/UEGitPlugin#146] [Q&A] [answered]
Hello, I'm fairly confused about how this plugin works with feature branches. Me and another developer wants to have a main branch and we create feature branches per new story we have. How does this plugin work with if I am on branch XX-1 and he's on XX-2? How would we lock uasset files from each other? Or does this plugin not work that way and we need to work off of a single branch?

**Comment by mastercoms** [comment]
Status branches support wildcards:

* origin/main
* origin/feature/*

Status branches check for differences between branches that have already been pushed and unlocked. I'd recommend you to keep a branch format like this for your story IDs, so you can distinguish between any mainline/release branches and your feature branches.

And LFS locks (while they are active) are always cross branch.

**Reply by JZWong** [reply]
Are you saying this for the code?

`const TArray<FString> Branches {"origin/main", "origin/feature/*"};`

Currently our branch names are just ZS-1

**Reply by mastercoms** [reply]
Yes, I can add some code in the next release to support wildcarding things that will match already specified branches, if you do not want to adjust your branch naming scheme.

**Reply by JZWong** [reply]
Sorry to be annoying. So you're saying with the status branches, my friend should see a lock on branch ZS-2 when I lock a file on ZS-1? I also went with `const TArray<FString> Branches {"origin/main", "origin/ZS-*"};`

**Reply by mastercoms** [reply, marked as answer]
You're not being annoying, it's okay! Locking a file will work regardless of status branches. You should see a lock on ZS-1 in ZS-2 even without status branches. Unlocking a file after it's pushed (default automatic behavior) is where status branches come in. It will check for differences between branches and modifications until those modifications are merged in. Also I think you don't need a double * for your branch there.

**Reply by JZWong** [reply]
Oh gotcha. That's awesome. My other developer hasn't hopped on to check if that is how it works, so I will wait until he gets on to see that, but that's really good to know. Thanks! I may or may not have more questions :D
